### PR TITLE
fix(TDC-7310) - fixed select all checkbox disappearing

### PR DESCRIPTION
**Pull request by dlcaldeira**
**What is the problem this PR is trying to solve?**
"Select all" checkbox was not showing

**What is the chosen solution to this problem?**
Replaced checkbox with TUI checkbox

**Please check if the PR fulfills these requirements**

- [x] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/List/Toolbar/SelectAll/SelectAll.component.js [hunk 1]
```diff
@@ -1,32 +1,29 @@
...
-							onToggleAll(event, items);
-						}}
-						checked={isAllSelected()}
-						disabled={!items.length}
-					/>
-					<span>{t('LIST_SELECT_ALL', { defaultValue: 'Select all' })}</span>
-				</label>
-			</div>
+		<form className={theme.container}>
+			<label htmlFor={checkboxId}>
```
Comment (jmfrancois) [inline packages/components/src/List/Toolbar/SelectAll/SelectAll.component.js:15]: this duplicate the label there is already one inside `Form.Checkbox`

File: packages/components/src/List/Toolbar/SelectAll/__snapshots__/SelectAll.snapshot.test.js.snap [hunk 2]
```diff
@@ -2,74 +2,52 @@
...
-        onChange={[Function]}
-        type="checkbox"
-      />
-      <span>
-        Select all
-      </span>
-    </label>
-  </div>
+  <label>
+    <span
```
Comment (jmfrancois) [inline packages/components/src/List/Toolbar/SelectAll/__snapshots__/SelectAll.snapshot.test.js.snap:8]: if you want a better snap you should use a jest.unmock('@talend/design-system')` at the begining of the test

**Review by jmfrancois — CHANGES_REQUESTED** [review]
please fix the comment

File: packages/components/src/List/Toolbar/SelectAll/SelectAll.component.js (hunk 1 above)
Comment (dlcaldeira) [reply packages/components/src/List/Toolbar/SelectAll/SelectAll.component.js:15]: Youre totally right 👌

**Review by jmfrancois — APPROVED** [review]

**Review by romainseb — APPROVED** [review]

**Pull request merged**